defi/src/utils/shared/distressedCoins.ts

```diff
@@ -1,32 +1,91 @@
-import { storeR2JSONString, getR2JSONString } from "../../utils/r2";
+import { getUnixTimeNow } from "../../api2/utils/time";
 import { chainsThatShouldNotBeLowerCased } from "../../utils/shared/constants";
+import { elastic, cache } from "@defillama/sdk";
 
-const r2Key = "distressedAssetsList.json";
-
-export async function isDistressed(key: string) {
+function sanitizeKey(key: string) {
   const chain = key.split(":")[0];
   const address = key.substring(chain.length + 1);
-  const normalizedAddress = chainsThatShouldNotBeLowerCased.includes(chain)
-    ? address
-    : address.toLowerCase();
-  const data = await getR2JSONString(r2Key);
+  const normalizedAddress = chainsThatShouldNotBeLowerCased.includes(chain) ? address : address.toLowerCase();
+  return `${chain}:${normalizedAddress}`;
+}
+
+export async function isDistressed(key: string, client?: any) {
+  const isLocalClient: boolean = client == undefined
+  if (isLocalClient) client = elastic.getClient();
 
-  if (!data[chain]) return false;
-  if (data[chain][normalizedAddress]) return true;
+  const _id = sanitizeKey(key)
+  const { hits } = await client.search({
+    index: "distressed-assets-store*",
+    body: {
+      query: {
+        match: { _id },
+      },
+    },
+  });
 
-  return false;
+  if (isLocalClient) await client?.close();
```

Comment:
can leave the connection open, can close it once when process exits, noticed this pattern in other places as well, not sure if it is a good idea to open & close connection per requests, most of our scripts run under an hour, so dont see why we need to close it per write